### PR TITLE
Add Manufacturer Data to advertisement.

### PR DESCRIPTION
**Pull request by Jojo134**
Sets the manufacturer data in an advertisement. Added a call to this method also for beacons.

### CODE REVIEW

File: bluezero/advertisement.py [hunk 1]
```diff
@@ -123,9 +123,13 @@ def service_UUIDs(self, UUID):  # pylint: disable=invalid-name
                  'ServiceUUIDs',
                  UUID)
 
-    def manufacturer_data(self):
-        """Manufacturer Data to be broadcast (Currently not supported)"""
-        pass
+    def manufacturer_data(self, company_id, data):
+        """Manufacturer Data to be broadcast
+
+        ad.manufacturer_data({int("0x004c", 16):bytearray.fromhex(23*"AA")})"""
```
Comment (ukBaz) [inline bluezero/advertisement.py:129]: This doesn't feel list a very helpful docstring as I wouldn't expect anyone to assign the values that way.
Can we also not use the Apple Inc Company Identifier of 0x004c.

File: bluezero/broadcaster.py [hunk 2]
```diff
@@ -40,9 +40,13 @@ def add_manufacturer_data(self, manufacturer, data):
         Add manufacturer information to be used in beacon message
         :param manufacturer: Use numbers from Bluetooth SIG
         https://www.bluetooth.com/specifications/assigned-numbers/16-bit-UUIDs-for-Members
-        :param data: Data to be sent (Limit of ??)
+
+        :param data: Data to be sent (Limit of 23bytes)
```
Comment (ukBaz) [inline bluezero/broadcaster.py:44]: `bytearray.fromhex(23*"FF")` doesn't feel right to me I would prefer something like: `b'\xff' * 23` if you want to keep the line length manageable.

**Review by ukBaz — COMMENTED** [review]
Couple of minor points about the docstrings but apart from that looks good.

Do you want to add an example to this PR?
We need to avoid iBeacon so use https://altbeacon.org/ 

Not sure how to best represent the long manufacturing data line. I have had a go three different ways. Do you have a preference?

```python
from bluezero import broadcaster


def main():
    alt = broadcaster.Beacon()
    alt.add_manufacturer_data(
        0xffff,  # Manufacturer ID (0xffff = Not for production)
        b'\xBE\xAC'  # beacon code for Alt Beacon
        b'\x22\x22\x22\x22\x22\x22\x22\x22\x22\x22\x22\x22\x22\x22\x22\x22'
        b'\x00\x00\x00\x00'  # Free 
        b'\x01')  # Transmit power
    alt.add_manufacturer_data('ffff',  # Manufacturer ID (0xffff = Not for production)
                              b'\xBE\xAC'   # beacon code for Alt Beacon
                              + 16 * b'\xbe'   # Beacon UUID
                              + 4 * b'\x00'  # Free
                              + b'\x01')  # Transmit power
    alt.add_manufacturer_data(
        65535,  # Manufacturer ID (0xffff = Not for production)
        [190, 172,   # beacon code for Alt beacon
         34, 34, 34, 34, 34, 34, 34, 34, 34, 34, 34, 34, 34, 34, 34, 34,
         0, 0, 0, 0,  # Free
         1])  # Transmit power
    alt.start_beacon()


if __name__ == '__main__':
    main()
```

File: bluezero/advertisement.py (hunk 1 above)
Comment (Jojo134) [reply bluezero/advertisement.py]: yes i removed this. I hope with the example it should be clear

**Comment by Jojo134** [comment]
I went with the second form in the example since one can easy see the amount of bytes without having to count them. And adapt parts for beacons with a different advertisement format.

**Pull request merged**